code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -352,6 +367,12 @@ var/list/gaslist_cache = null
 		removed.add_gas(id)
 		removed_gases[id][MOLES] = QUANTIZE((cached_gases[id][MOLES] / sum) * amount)
 		cached_gases[id][MOLES] -= removed_gases[id][MOLES]
+	for(var/R in cached_reagents)
+		var/datum/reagent/RA = R
+		var/temp_id = RA.id
+		var/mathmemes = QUANTIZE((RA.volume / reagents.total_volume) * amount)

```

Comment:
doesn't matter, it's just an optimization pass, this would just be datum/reagent/R in the for loop if it wasnt so laggy